core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -337,15 +340,20 @@ private CloseableIterable<ManifestEntry<DataFile>> addedDataFiles(
       TableMetadata base,
       Long startingSnapshotId,
       Expression dataFilter,
-      PartitionSet partitionSet) {
+      PartitionSet partitionSet,
+      Snapshot parent) {
     // if there is no current table state, no files have been added
-    if (base.currentSnapshot() == null) {
+    if (parent == null) {
```

Comment:
My bad, missed this when I raised the change to your branch. I think we should update this comment to say "If there is no parent, no files have been added"